tests/lib/index.js

```diff
@@ -1063,5 +1063,445 @@ describe('config', function () {
+            });
+          });
+
+          it("fails on unknown config", function(next) {
+            var config = new Config();
+            config
+              .addConfigPromise(
+                "modown-newsboxes",
+                "application",
+                libpath.resolve(mojito, "application.json")
+              )
+              .then(() =>
+                config
+                  .readNoMergePromise("modown-newsboxes", "foo", {})
+                  .catch(err => {
+                    expect(err.message).to.equal(
+                      'Unknown config "foo" in bundle "modown-newsboxes"'
+                    );
+                    next();
+                  })
+              );
+          });
+
+          it("reads non-contextualized .json config files", function(next) {
+            var config = new Config();
+            config
+              .addConfigPromise(
+                "simple",
+                "routes",
+                libpath.resolve(touchdown, "configs/dimensions.json")
+              )
+              .then(() =>
+                config.readNoMergePromise("simple", "routes", {}).then(have => {
+                  expect(have).to.be.an("array");
+                  expect(have[0]).to.be.an("array");
+                  expect(have[0][0]).to.be.an("object");
+                  expect(have[0][0].dimensions).to.be.an("array");
+                  next();
+                })
+              );
+          });
+
+          it("reads contextualized .js config files", function(next) {
+            var config = new Config();
+            config
+              .addConfigPromise(
+                "simple",
+                "dimensions",
+                libpath.resolve(touchdown, "configs/dimensions.json")
+              )
+              .then(() =>
+                config
+                  .addConfigPromise(
+                    "simple",
+                    "foo",
+                    libpath.resolve(touchdown, "configs/foo.js")
+                  )
+                  .then(() =>
+                    config
+                      .readNoMergePromise("simple", "foo", { device: "mobile" })
+                      .then(have => {
+                        expect(have).to.be.an("array");
+                        expect(have[0]).to.be.an("object");
+                        expect(have[0].TODO).to.equal("TODO");
+                        expect(have[1]).to.be.an("object");
+                        expect(have[1].selector).to.equal("mobile");
+                        next();
+                      })
+                  )
+              );
+          });
+
+          it("reads contextualized .json config files", function(next) {
+            var config = new Config();
+            config
+              .addConfigPromise(
+                "modown",
+                "dimensions",
+                libpath.resolve(mojito, "node_modules/modown/dimensions.json")
+              )
+              .then(() =>
+                config
```

Comment:
Actually, I intentionally return promise directly, we could align the `then` or `catch` outside in the same level if you want.

Reply:
Ya that would be better then  the  nested  promise  handling and allow us to use async await more easily  when we drop node 8 support.

Reply:
Agreed.